File: frontend/src/lib/github.ts
import { apiDelete, apiGet, apiPost } from "@/lib/api";

export interface GitHubAccount {
  connected: boolean;
  username: string | null;
  email: string | null;
  avatar_url: string | null;
  repo_full_name: string | null;
  repo_default_branch: string | null;
  base_dir: string | null;
  last_sync_at: string | null;
}

export interface GitHubRepo {
  full_name: string;
  private: boolean;
  default_branch: string;
  clone_url: string;
  updated_at: string;
}

export interface GitHubBranch {
  name: string;
}

export const githubStatus = (): Promise<GitHubAccount> => apiGet("/github/status");

export const githubConnect = (token: string): Promise<GitHubAccount> =>
  apiPost("/github/auth", { token });

export const githubDisconnect = (): Promise<void> => apiDelete("/github/auth");

export const githubRepos = (): Promise<GitHubRepo[]> => apiGet("/github/repos");

export const githubBranches = (repoFullName: string): Promise<GitHubBranch[]> =>
  apiGet(`/github/repos/${repoFullName}/branches`);

File: backend/src/nbplatform/schemas/github.py
"""Schemas da integração GitHub (conta pessoal do usuário)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class GitHubConnectRequest(BaseModel):
    token: str = Field(min_length=1, max_length=500)


class GitHubAccountRead(BaseModel):
    connected: bool
    username: str | None = None
    email: str | None = None
    avatar_url: str | None = None
    repo_full_name: str | None = None
    repo_default_branch: str | None = None
    base_dir: str | None = None
    last_sync_at: datetime | None = None


class GitHubRepoRead(BaseModel):
    full_name: str
    private: bool
    default_branch: str
    clone_url: str
    updated_at: str


class GitHubBranchRead(BaseModel):
    name: str
